core/jvm/src/main/scala/zio/internal/PlatformSpecific.scala

```diff
@@ -38,6 +40,13 @@ private[internal] trait PlatformSpecific {
       }
     }
 
+  /**
+   * Adds a signal handler that executes the specified action on receiving a signal.
+   */
+  def addSignalHandler(signal: String)(handler: () => Unit): Unit = {
```

Comment:
This is a pretty unsafe method as we saw earlier because if a signal handler has already been installed for the signal this will throw. What if we catch the error here and change the function to return a boolean indicating either true if we were able to successfully install the signal hander or false if another signal handler was already installed?